# nekrs.hypregpu: exact upstream (96b3cf9e5bacede16568826c04a21bc0fe50dc7d) -> frozen baseline; generated by freeze-1.0 from the applied patch series
--- original/src/cmake/hypre.cmake
+++ baseline/src/cmake/hypre.cmake
@@ -58,7 +58,7 @@
   set(HYPRE_BACKEND "--with-cuda" "--with-cuda-home=${CUDAToolkit_LIBRARY_ROOT}")
 
   if(CUDAToolkit_VERSION VERSION_GREATER_EQUAL "13.0.0")
-    set(HYPRE_DEVICE_ARCH "HYPRE_CUDA_SM=80 90")
+    set(HYPRE_DEVICE_ARCH "HYPRE_CUDA_SM=80 90 100")
     #disable for now as it might not play well with all MPI implementations
     #set(HYPRE_CONFIGURE_FLAGS "--enable-device-malloc-async")
   elseif(CUDAToolkit_VERSION VERSION_GREATER_EQUAL "12.0.0")
--- original/src/3rd_party/hypre/src/seq_mv/csr_matop_device.c
+++ baseline/src/3rd_party/hypre/src/seq_mv/csr_matop_device.c
@@ -1034,7 +1034,7 @@
    hypre_assert(new_end.first - reduced_col_indices == new_end.second - reduced_col_nnz);
    num_reduced_col_indices = new_end.first - reduced_col_indices;
 #else
-   thrust::pair<HYPRE_Int*, HYPRE_Int*> new_end =
+   auto new_end =
       HYPRE_THRUST_CALL(reduce_by_key, A_j_sorted, A_j_sorted + nnz_A,
                         thrust::make_constant_iterator(1),
                         reduced_col_indices,
--- original/src/3rd_party/hypre/src/utilities/device_utils.c
+++ baseline/src/3rd_party/hypre/src/utilities/device_utils.c
@@ -1491,7 +1491,7 @@
 #else
       HYPRE_THRUST_CALL(sort_by_key, map2, map2 + ny, y);
 
-      thrust::pair<HYPRE_Int*, HYPRE_Real*> new_end = HYPRE_THRUST_CALL( reduce_by_key,
+      auto new_end = HYPRE_THRUST_CALL( reduce_by_key,
                                                                          map2,
                                                                          map2 + ny,
                                                                          y,
--- original/src/3rd_party/hypre/src/IJ_mv/IJMatrix_parcsr_device.c
+++ baseline/src/3rd_party/hypre/src/IJ_mv/IJMatrix_parcsr_device.c
@@ -656,7 +656,7 @@
                             is_on_proc,                                                                                                               /* stencil */
                             thrust::make_zip_iterator(thrust::make_tuple(off_proc_i,      off_proc_j,      off_proc_data,
                                                                          off_proc_sora)),       /* result */
-                            thrust::not1(thrust::identity<char>()) );
+                            thrust::not_fn(thrust::identity<char>()) );
 
          hypre_assert(thrust::get<0>(new_end1.get_iterator_tuple()) - off_proc_i == nelms_off);
 
@@ -668,7 +668,7 @@
                             thrust::make_zip_iterator(thrust::make_tuple(stack_i + nelms, stack_j + nelms, stack_data + nelms,
                                                                          stack_sora + nelms)),  /* last */
                             is_on_proc,                                                                                                               /* stencil */
-                            thrust::not1(thrust::identity<char>()) );
+                            thrust::not_fn(thrust::identity<char>()) );
 
          hypre_assert(thrust::get<0>(new_end2.get_iterator_tuple()) - stack_i == nelms_on);
 #endif
--- original/src/3rd_party/hypre/src/IJ_mv/IJVector_parcsr_device.c
+++ baseline/src/3rd_party/hypre/src/IJ_mv/IJVector_parcsr_device.c
@@ -522,7 +522,7 @@
                             is_on_proc,                                                         /* stencil */
                             thrust::make_zip_iterator(thrust::make_tuple(off_proc_i,      off_proc_data,
                                                                          off_proc_sora)),       /* result */
-                            thrust::not1(thrust::identity<char>()) );
+                            thrust::not_fn(thrust::identity<char>()) );
 
          hypre_assert(thrust::get<0>(new_end1.get_iterator_tuple()) - off_proc_i == nelms_off);
 
@@ -534,7 +534,7 @@
                             thrust::make_zip_iterator(thrust::make_tuple(stack_i + nelms, stack_data + nelms,
                                                                          stack_sora + nelms)),  /* last */
                             is_on_proc,                                                         /* stencil */
-                            thrust::not1(thrust::identity<char>()) );
+                            thrust::not_fn(thrust::identity<char>()) );
 
          hypre_assert(thrust::get<0>(new_end2.get_iterator_tuple()) - stack_i == nelms_on);
 #endif
--- original/src/3rd_party/hypre/src/parcsr_ls/ams.c
+++ baseline/src/3rd_party/hypre/src/parcsr_ls/ams.c
@@ -697,7 +697,7 @@
          1.0 );
 #else
          thrust::identity<HYPRE_Complex> identity;
-         HYPRE_THRUST_CALL( replace_if, l1_norm, l1_norm + num_rows, thrust::not1(identity), 1.0 );
+         HYPRE_THRUST_CALL( replace_if, l1_norm, l1_norm + num_rows, thrust::not_fn(identity), 1.0 );
 #endif
       }
       else
@@ -768,7 +768,7 @@
       HYPRE_THRUST_CALL( transform_if, l1_norm, l1_norm + num_rows, diag_tmp, l1_norm,
                          thrust::negate<HYPRE_Real>(),
                          is_negative<HYPRE_Real>() );
-      //bool any_zero = HYPRE_THRUST_CALL( any_of, l1_norm, l1_norm + num_rows, thrust::not1(thrust::identity<HYPRE_Complex>()) );
+      //bool any_zero = HYPRE_THRUST_CALL( any_of, l1_norm, l1_norm + num_rows, thrust::not_fn(thrust::identity<HYPRE_Complex>()) );
       bool any_zero = 0.0 == HYPRE_THRUST_CALL( reduce, l1_norm, l1_norm + num_rows, 1.0,
                                                 thrust::minimum<HYPRE_Real>() );
 #endif
--- original/src/3rd_party/hypre/src/parcsr_ls/par_mod_multi_interp_device.c
+++ baseline/src/3rd_party/hypre/src/parcsr_ls/par_mod_multi_interp_device.c
@@ -327,7 +327,7 @@
                                               thrust::make_counting_iterator(n_fine),
                                               CF_marker,
                                               points_left,
-                                              thrust::not1(equal<HYPRE_Int>(1)) );
+                                              thrust::not_fn(equal<HYPRE_Int>(1)) );
    remaining = points_end - points_left;
 
    /* Cpts; number of C pts */
@@ -541,7 +541,7 @@
                                       points_left_old + remaining,
                                       diag_shifts,
                                       points_left,
-                                      thrust::not1(thrust::identity<HYPRE_Int>()) );
+                                      thrust::not_fn(thrust::identity<HYPRE_Int>()) );
 #endif
 
          hypre_assert(new_end - points_left == cnt_rem);
@@ -1637,7 +1637,7 @@
                       fine_to_coarse,
                       fine_to_coarse + n_fine,
                       pass_marker,
-                      thrust::not1(equal<HYPRE_Int>(color)),
+                      thrust::not_fn(equal<HYPRE_Int>(color)),
                       -1 );
 #endif
 }
--- original/src/3rd_party/hypre/src/parcsr_mv/par_csr_triplemat_device.c
+++ baseline/src/3rd_party/hypre/src/parcsr_mv/par_csr_triplemat_device.c
@@ -229,7 +229,7 @@
                                                                 hypre_CSRMatrixData(Cbar))) + hypre_CSRMatrixNumNonzeros(Cbar),
                    hypre_CSRMatrixJ(Cbar),
                    thrust::make_zip_iterator(thrust::make_tuple(C_offd_ii, C_offd_j, C_offd_a)),
-                   thrust::not1(pred) );
+                   thrust::not_fn(pred) );
       hypre_assert( thrust::get<0>(new_end.get_iterator_tuple()) == C_offd_ii + nnz_C_offd );
 #endif
 
@@ -958,7 +958,7 @@
                                                                                      Cext_bigj)) + Cext_nnz,
                                         Cext_bigj,
                                         thrust::make_zip_iterator(thrust::make_tuple(work, big_work)),
-                                        thrust::not1(pred1) );
+                                        thrust::not_fn(pred1) );
 
       HYPRE_Int Cext_offd_nnz = thrust::get<0>(off_end.get_iterator_tuple()) - work;
 #endif
@@ -1220,7 +1220,7 @@
                                 thrust::make_zip_iterator(thrust::make_tuple(zmp_i, zmp_j, zmp_a)) + local_nnz_C,
                                 zmp_j,
                                 thrust::make_zip_iterator(thrust::make_tuple(C_offd_ii, C_offd_j, C_offd_a)),
-                                thrust::not1(pred) );
+                                thrust::not_fn(pred) );
    hypre_assert( thrust::get<0>(new_end.get_iterator_tuple()) == C_offd_ii + nnz_C_offd );
 #endif
    hypreDevice_CsrRowIndicesToPtrs_v2(hypre_CSRMatrixNumRows(C_offd), nnz_C_offd, C_offd_ii,
--- original/src/3rd_party/hypre/src/seq_mv/csr_matop_device.c
+++ baseline/src/3rd_party/hypre/src/seq_mv/csr_matop_device.c
@@ -681,7 +681,7 @@
                         B_ext_bigj,                                                            /* stencil */
                         thrust::make_zip_iterator(thrust::make_tuple(B_ext_offd_ii, B_ext_offd_bigj, B_ext_offd_data,
                                                                      B_ext_offd_xata)),        /* result */
-                        thrust::not1(pred1) );
+                        thrust::not_fn(pred1) );
 
       hypre_assert( thrust::get<0>(new_end.get_iterator_tuple()) == B_ext_offd_ii + B_ext_offd_nnz );
 #endif
@@ -708,7 +708,7 @@
                         B_ext_bigj,                                                            /* stencil */
                         thrust::make_zip_iterator(thrust::make_tuple(B_ext_offd_ii, B_ext_offd_bigj,
                                                                      B_ext_offd_data)),        /* result */
-                        thrust::not1(pred1) );
+                        thrust::not_fn(pred1) );
 
       hypre_assert( thrust::get<0>(new_end.get_iterator_tuple()) == B_ext_offd_ii + B_ext_offd_nnz );
 #endif
@@ -2145,7 +2145,7 @@
       new_nnz = HYPRE_THRUST_CALL( count_if,
                                    A_data,
                                    A_data + nnz,
-                                   thrust::not1(less_than<HYPRE_Complex>(tol)) );
+                                   thrust::not_fn(less_than<HYPRE_Complex>(tol)) );
    }
    else
    {
@@ -2199,7 +2199,7 @@
                                         thrust::make_zip_iterator(thrust::make_tuple(A_ii, A_j, A_data)) + nnz,
                                         A_data,
                                         thrust::make_zip_iterator(thrust::make_tuple(new_ii, new_j, new_data)),
-                                        thrust::not1(less_than<HYPRE_Complex>(tol)) );
+                                        thrust::not_fn(less_than<HYPRE_Complex>(tol)) );
 
       hypre_assert( thrust::get<0>(new_end.get_iterator_tuple()) == new_ii + new_nnz );
    }
--- original/src/3rd_party/hypre/src/utilities/_hypre_utilities.hpp
+++ baseline/src/3rd_party/hypre/src/utilities/_hypre_utilities.hpp
@@ -359,6 +359,8 @@
 #include <thrust/iterator/constant_iterator.h>
 #include <thrust/iterator/counting_iterator.h>
 #include <thrust/iterator/zip_iterator.h>
+#include <thrust/iterator/reverse_iterator.h>
+#include <thrust/pair.h>
 #include <thrust/transform.h>
 #include <thrust/functional.h>
 #include <thrust/gather.h>
--- original/src/3rd_party/hypre/src/utilities/device_utils.h
+++ baseline/src/3rd_party/hypre/src/utilities/device_utils.h
@@ -164,6 +164,8 @@
 #include <thrust/iterator/constant_iterator.h>
 #include <thrust/iterator/counting_iterator.h>
 #include <thrust/iterator/zip_iterator.h>
+#include <thrust/iterator/reverse_iterator.h>
+#include <thrust/pair.h>
 #include <thrust/transform.h>
 #include <thrust/functional.h>
 #include <thrust/gather.h>
